giskard/client/giskard_client.py

```diff
@@ -180,7 +183,41 @@ def get_project(self, project_key: str) -> Project:
         response = self._session.get("project", params={"key": project_key}).json()
         return Project(self._session, response["key"], response["id"])
 
-    def create_project(self, project_key: str, name: str, kernel_name: str, description: str = None) -> Project:
+    def initialize_kernel(self, project_key: str):
+        python_version = f"{sys.version_info[0]}.{sys.version_info[1]}"
+        kernel_name = f"{project_key}_kernel"
+
+        kernels = self._session.get("kernels").json()
+        frozen_dependencies = [f"{dist.name}=={dist.version}" for dist in importlib_metadata.distributions()]
+
+        existing_kernel = next((kernel for kernel in kernels if kernel["name"] == kernel_name), None)
+
+        if (
+            existing_kernel
+            and existing_kernel["pythonVersion"] == python_version
+            and set(frozen_dependencies).issubset(set(existing_kernel["frozenDependencies"].split("\n")))
+        ):
+            # A similar kernel already exists
+            return existing_kernel["name"]
+        elif existing_kernel:
+            # A different kernel exists that uses the same name
+            kernel_name = f"{kernel_name}_{uuid.uuid4()}"
+
+        self._session.post(
```

Comment:
Can we extract this into a method like `create_kernel` in Giskard client?

I am also trying to do something similar for list kernels. Otherwise, I can do it on my side after merging this branch.

Reply:
Sure it's done